core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingSimulator.kt

```diff
@@ -13,51 +14,97 @@ import fr.sncf.osrd.utils.units.Offset
 interface ETCSBrakingSimulator {
     val context: EnvelopeSimContext
 
-    /** Compute the ETCS braking envelope for each LOA. */
+    /**
+     * Compute the ETCS braking envelope for each LoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addSlowdownBrakingCurves(
         envelope: Envelope,
         limitsOfAuthority: Collection<LimitOfAuthority>
     ): Envelope
 
-    /** Compute the ETCS braking envelope for each EOA. */
+    /**
+     * Compute the ETCS braking envelope for each EoA and return a new envelope taking said curves
+     * into account.
+     */
     fun addStopBrakingCurves(
         envelope: Envelope,
         endsOfAuthority: Collection<EndOfAuthority>
     ): Envelope
+
+    /** Compute the ETCS braking curves for each LoA, ordered by LoA offset. */
+    fun computeSlowdownBrakingCurves(
+        mrsp: Envelope,
+        limitsOfAuthority: Collection<LimitOfAuthority>
+    ): LOAETCSBrakingCurves
+
+    /** Compute the ETCS braking curves for each EoA, ordered by EoA offset. */
+    fun computeStopBrakingCurves(
+        mrsp: Envelope,
+        endsOfAuthority: Collection<EndOfAuthority>
+    ): EOAETCSBrakingCurves
 }
 
+typealias ETCSBrakingCurves = EnumMap<BrakingCurveType, ETCSBrakingCurve?>
```

Comment:
nit: If `BrakingCurveType` is not planned to contain `CONST`, it should probably be renamed.
```suggestion
typealias ETCSBrakingCurves = EnumMap<ETCSBrakingCurveType, ETCSBrakingCurve?>
```
If it is ever planned to contain `CONST`, then we can get rid of all the `ETCS` 🎉 
```suggestion
typealias BrakingCurves = EnumMap<BrakingCurveType, BrakingCurve?>
```

Reply:
Honestly? Yeah, it'll make the code cleaner, I'll combine both and handle all the other cases in `TrainPhysicsIntegrator.java` in an exception. Might as well rename it simply to `BrakingType` and so on.